kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/filter/authorization/TxnOffsetCommitAuthzIT.java

```diff
@@ -0,0 +1,422 @@
+            Response response = client.getSync(new Request(ApiKeys.JOIN_GROUP, (short) 9, "admin", request));
+            assertThat(response.payload().message()).isInstanceOfSatisfying(JoinGroupResponseData.class, responseData -> {
+                if (responseData.memberId() != null && !responseData.memberId().isEmpty()) {
+                    memberId.set(responseData.memberId());
+                }
+                generationId.set(responseData.generationId());
+                assertThat(Errors.forCode(responseData.errorCode())).isEqualTo(Errors.NONE);
+            });
+        });
+        return new MemberIdAndGeneration(memberId.get(), generationId.get());
+    }
+
+    private static ProducerIdAndEpoch initProducerId(KafkaClient client, String transactionalId) {
+        AtomicReference<ProducerIdAndEpoch> producerIdAndEpoch = new AtomicReference<>(ProducerIdAndEpoch.NONE);
+        Awaitility.await().untilAsserted(() -> {
+            InitProducerIdRequestData request = new InitProducerIdRequestData();
+            request.setTransactionalId(transactionalId);
+            ProducerIdAndEpoch pep = producerIdAndEpoch.get();
+            request.setProducerId(pep.producerId);
+            request.setProducerEpoch(pep.epoch);
+            request.setTransactionTimeoutMs(10000);
+            Response response = client.getSync(new Request(ApiKeys.INIT_PRODUCER_ID, (short) 5, "admin", request));
+            assertThat(response.payload().message()).isInstanceOfSatisfying(InitProducerIdResponseData.class, responseData -> {
+                assertThat(Errors.forCode(responseData.errorCode())).isEqualTo(Errors.NONE);
+                producerIdAndEpoch.set(new ProducerIdAndEpoch(responseData.producerId(), responseData.producerEpoch()));
+            });
+        });
+        return producerIdAndEpoch.get();
+    }
+
+    private static void findCoordinator(FindCoordinatorRequest.CoordinatorType transaction, String transactionalId, KafkaClient client) {
+        Awaitility.await().untilAsserted(() -> {
+            FindCoordinatorRequestData request = new FindCoordinatorRequestData();
+            FindCoordinatorRequest.CoordinatorType type = transaction;
+            request.setKeyType(type.id());
+            request.coordinatorKeys().add(transactionalId);
+            Response response = client.getSync(new Request(ApiKeys.FIND_COORDINATOR, (short) 6, "admin", request));
+            assertThat(response.payload().message()).isInstanceOfSatisfying(FindCoordinatorResponseData.class, responseData -> {
+                assertThat(Errors.forCode(responseData.errorCode())).isEqualTo(Errors.NONE);
+                responseData.coordinators().forEach(coordinator -> {
+                    assertThat(Errors.forCode(coordinator.errorCode())).isEqualTo(Errors.NONE);
+                });
+            });
+        });
+    }
+
+    private static void saslPlainAuthenticate(KafkaClient client, String username, String password) {
```

Comment:
There's code in the base class for authentication. Is there some reason why we can't reuse that?

Reply:
have changed to use the same style as the OffsetCommit test, using the authenticated clients.